subsys/spm/secure_services_ns.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <zephyr.h>
+#include <errno.h>
+#include <../arch/arm/include/aarch32/cortex_m/tz_ns.h>
+#include <fw_info.h>
+
+void spm_request_system_reboot_nsc(void);
+int spm_request_random_number_nsc(uint8_t *output, size_t len, size_t *olen);
+int spm_request_read_nsc(void *destination, uint32_t addr, size_t len);
+int spm_firmware_info_nsc(uint32_t fw_address, struct fw_info *info);
+int spm_prevalidate_b1_upgrade_nsc(uint32_t dst_addr, uint32_t src_addr);
+void spm_busy_wait_nsc(uint32_t busy_wait_us);
+
+TZ_THREAD_SAFE_NONSECURE_ENTRY_FUNC(spm_request_system_reboot,
```

Comment:
I guess that the commit message was composed before this functionality was available in zephyr, so you should check that it is up to date. 

```
call k_sched_lock before and k_sched_unlock after.

Rename the services from spm_foo() to spm_foo_nsc() and call
spm_foo_nsc() from spm_foo() along with k_sched_*()
```

There is no mention of `k_sched` in this PR since it is all handled in the TZ macros.

Reply:
and by the way, you should elaborate here why THREAD_SAFE is needed (maybe some simple inline comments above the definitions)

Reply:
Reworded the commit message. I didn't add anything about thread safety here, but I added docs elsewhere.